django/db/models/expressions.py

```diff
@@ -412,6 +432,12 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
     def refs_aggregate(self, existing_aggregates):
         return refs_aggregate(self.name.split(LOOKUP_SEP), existing_aggregates)
 
+    def asc(self):
```

Comment:
`F` is a weird example, but you've pointed out potential others which leads me to believe there may be a need. 

I'm almost of the opinion that `F` is being too clever by not implementing the full expressions API, and we shouldn't break the relative purity of Combinable to save a few outliers from copying a couple of methods.

I brought this up when you first changed F to not be an expression - but do we need to have 2 separate concepts? Resolvable and Expressions. Expressions implement Resolvable, and Resolvables resolve to Expressions.

Resolvables would have `resolve_expression` plus any other methods that could be called before being resolved to full expressions.

Then the definition of `F` becomes `class F(CombinableMixin, Resolvable)`. It's a bit more of a concept to grasp, but the majority of expressions will derive from ExpressionNode rather than the two mixins above.